### ISSUE TITLE
Possible Error in \b Explanation

### ISSUE THREAD

**Issue by sjordan1975** [firasdib/Regex101#785]
I received an answer to a question I raised on Stackover which suggests a bug in at least the Explanation of one of the Regex

Regex101.com Explanation suggests:
`\b is shorthand for (^\w|\w$|\W\w|\w\W)AS(^\w|\w$|\W\w|\w\W)`

SO answer that suggests that is not the case: 
http://stackoverflow.com/a/42239224/471249

**Comment by firasdib** [comment]
The explanation is trying to tell you where a word boundary exists, not that its a shorthand for that.

**Comment by sjordan1975** [comment]
@firasdib Thank you for speedy reply; great to hear from the owner of the project.

I do believe the explanation is a bit confusing.

For example, type \b in the input box (and remove the /g modifier) and you get the following explanation:
`\b assert position at a word boundary (^\w|\w$|\W\w|\w\W)`

Type ^ in the input box and you get the following explanation:
`^ asserts position at start of the string`

There's inconsistency in the explanations. 

By including _(^\w|\w$|\W\w|\w\W)_ in the first explanation immediately after the word "boundary", it implies that it's a substitute for word boundary, which it is not.

Hope it help

Best

**Comment by Ouims** [comment]
As mentioned, the explanation is 'trying' to tell the user where the boundaries exist, it is not explicitely saying that \b is an equivalency to (^\w|\w$|\W\w|\w\W).
The /g modifier don't change the explanation of \b.
The ^ assertion has nothing to do with \b, the start of the string can hardly be confusing users, nobody has ever asked 'what is the start of the string', wheras regex's begginners often ask 'what is \b and what's a word boundary', I believe the extra explanation is there for that reason.

**Comment by Zarthus** [comment]
That explanation is confusing then, and should be amended to something more clear, as I was under the same impression.

**Comment by firasdib** [comment]
I'm open to suggestions.

**Comment by Zarthus** [comment]
Is it possible to add that info as a popover, and just explain \b is for zero width matching?

**Comment by TWiStErRob** [comment]
@Zarthus no need to worry about breaking the line: 
![image](https://cloud.githubusercontent.com/assets/2906988/23333160/bac2eac6-fb7e-11e6-96ac-9df5d8b9ab14.png)

**Comment by Zarthus** [comment]
no, but too lengthy explanations will just be annoying

**Comment by TWiStErRob** [comment]
current live version:
`\b` assert position at a word boundary `(^\w|\w$|\W\w|\w\W)`

I think it's good to define what a word boundary is, even if it builds on prior regex knowledge.

suggestions (**emphasis** mine, not to be included):
`\b` assert position at a word boundary (**i.e. near** `^\w|\w$|\W\w|\w\W`) *- to show the location of the matches*
`\b` is **approximately** a shorthand for `(^\w|\w$|\W\w|\w\W)` *- to express that it's like that, but not quite*

This conversation reminded me of this magical sketch: https://www.youtube.com/watch?v=_LC0JjvAJt8

**Comment by Zarthus** [comment]
I would consider "is indicative of" instead of "is approximately a shorthand for", and do away with the "i.e." in the first.

I think part of the problem stems from that Regex101 explains what \w and the likes are all for by showing what it actually is (like `[A-Za-z0-9_]`) and this is the only explanation that doesn't follow that pattern.

**Comment by TWiStErRob** [comment]
`i.e.` means "that is"/"read the prior as the following", I think it helps to qualify "near" so that it's clear that it is **explaining** what "word boundary" is.

Re pattern: all the others say "equal to", this one doesn't. Agree it needs some amending to have that non-qualification more explicit.

@sjordan1975 @Ouims Note that IMO [there's a better way of expressing this](https://regex101.com/r/HtwfCH/3), but it's too long and even more confusing than simply just omitting it (also this doesn't work in JS), we discarded this in a private conversation with Firas as it's too confusing.

**Comment by sjordan1975** [comment]
@Zarthus I totally agree: "this is the only explanation that doesn't follow that pattern". 

**THAT** was essentially my argument and reason I raised this; it breaks the explanation pattern. As a start, I would suggest removing `^\w|\w$|\W\w|\w\W` until wiser men than me come to consensus.

**Comment by firasdib** [comment]
I believe the following is good:

```
\b assert position at a word boundary: (^\w|\w$|\W\w|\w\W)
```


Simply adding a colon indicating where that word boundary is. Its not saying its equal to.

**Comment by Doqnach** [comment]
you are bound to plain text right? Otherwise drawing the same dashed line as you do in the test string to indicate a zero-width match might help clarify?

**Comment by Doqnach** [comment]
how about: `^:\w`, `\w:$`, `\W:\w` or `\w:\W`

Where `:` is the dashes line

**Comment by firasdib** [comment]
@Doqnach I found the dashes themselves to be the confusing part.

**Comment by firasdib** [comment]
![screenshot 2019-01-08 at 17 58 38](https://user-images.githubusercontent.com/1335165/50846279-0d7b6900-136f-11e9-822b-ba1c70bd3874.png)

It looks nicer with some air around it. What do you guys think?

**Comment by TWiStErRob** [comment]
I think it looks ok, it uses the same design language as an empty match, so it's consistent. I would halve the air though to let `|` stay the alternative separator.

**Comment by sjordan1975** [comment]
Awesome guys... I do appreciate that many in the discussion felt my raising the issue had value and collectively worked toward a solution!
